realm/realm-library/src/main/java/io/realm/internal/Collection.java

```diff
@@ -421,12 +431,12 @@ public boolean deleteLast() {
         if (observerPairs.isEmpty()) {
             nativeStartListening(nativePtr);
         }
-        CollectionObserverPair<T> collectionObserverPair = new CollectionObserverPair<T>(observer, listener);
+        CollectionObserverPair<T> collectionObserverPair = new CollectionObserverPair<>(observer, listener);
```

Comment:
My mind is a bit fuzzy on the details, but I believe because we support all the way back to API level 9, people can still use a Java 6 compiler. However since the annotation processor is compiled on our side, it should be possible for the annotation processor project to be set to 7 (or even 8). Only impact it would have is that people cloning the repo must have at least the same version (which they should, 7 has been EOL for a while now). Am I missing something?